secure-payment-confirmation/secure-payment-confirmation-authentication-accepted.https.html

```diff
@@ -0,0 +1,81 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Test for the 'secure-payment-confirmation' payment method authentication - accepted case</title>
+<link rel="help" href="https://w3c.github.io/secure-payment-confirmation#sctn-authentication">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<script src="utils.sub.js"></script>
+<script>
+'use strict';
+
+promise_test(async t => {
+  console.log('Adding authenticator');
+  const authenticator = await window.test_driver.add_virtual_authenticator(
+      AUTHENTICATOR_OPTS);
+  console.log('Authenticator:', authenticator);
+  t.add_cleanup(() => {
+    console.log('Clean-up: Removing authenticator');
+    return window.test_driver.remove_virtual_authenticator(authenticator);
+  });
+
+  console.log('Setting SPC Transaction Mode');
+  await window.test_driver.set_spc_transaction_mode("autoaccept");
+  t.add_cleanup(() => {
+    console.log('Resetting SPC Transaction Mode');
+    return window.test_driver.set_spc_transaction_mode("none");
+  });
+
+  console.log('Creating credential');
+  const credential = await createCredential();
```

Comment:
@nsatragno - Hey Nina, I'm having trouble with the WebAuthn part of these tests, and I'm hoping you might be able to help. Although they run correctly on my local system (Macbook), on the WPT CI the tests fail and it seems to be when trying to create a credential:

https://community-tc.services.mozilla.com/tasks/VYQcJnPnTDew3uRLx6Fysw/runs/0/logs/public/logs/live.log
```
 0:25.31 pid:2093 [2105:2105:1111/155745.145396:INFO:CONSOLE(14)] "Adding authenticator", source: https://web-platform.test:8443/secure-payment-confirmation/secure-payment-confirmation-authentication-accepted.https.html (14)
 0:25.41 pid:2093 [2105:2105:1111/155745.243308:INFO:CONSOLE(17)] "Authenticator: 584934ef-9af8-4516-9075-474c29c25a33", source: https://web-platform.test:8443/secure-payment-confirmation/secure-payment-confirmation-authentication-accepted.https.html (17)
 0:25.41 pid:2093 [2105:2105:1111/155745.244231:INFO:CONSOLE(23)] "Setting SPC Transaction Mode", source: https://web-platform.test:8443/secure-payment-confirmation/secure-payment-confirmation-authentication-accepted.https.html (23)
 0:25.42 pid:2093 [2105:2105:1111/155745.252374:INFO:CONSOLE(30)] "Creating credential", source: https://web-platform.test:8443/secure-payment-confirmation/secure-payment-confirmation-authentication-accepted.https.html (30)
 0:25.42 pid:2093 [2105:2105:1111/155745.253037:INFO:CONSOLE(18)] "createCredential, set_payment_extension: true", source: https://web-platform.test:8443/secure-payment-confirmation/utils.sub.js (18)
 0:25.43 pid:2093 [2105:2105:1111/155745.261787:INFO:CONSOLE(19)] "Clean-up: Removing authenticator", source: https://web-platform.test:8443/secure-payment-confirmation/secure-payment-confirmation-authentication-accepted.https.html (19)
 0:25.43 pid:2093 [2105:2105:1111/155745.261848:INFO:CONSOLE(26)] "Resetting SPC Transaction Mode", source: https://web-platform.test:8443/secure-payment-confirmation/secure-payment-confirmation-authentication-accepted.https.html (26)
 0:25.47 TEST_END: Test OK. Subtests passed 0/1. Unexpected 1
FAIL Successful SPC authentication - promise_test: Unhandled rejection with value: object "NotSupportedError: The user agent does not support public key credentials."
Error
    at get_stack (https://web-platform.test:8443/resources/testharness.js:3786:21)
    at new AssertionError (https://web-platform.test:8443/resources/testharness.js:3779:22)
    at assert (https://web-platform.test:8443/resources/testharness.js:3769:19)
    at Test.<anonymous> (https://web-platform.test:8443/resources/testharness.js:664:29)
    at Test.step (https://web-platform.test:8443/resources/testharness.js:2087:25)
    at https://web-platform.test:8443/resources/testharness.js:2114:35
```

Can you see anything obviously wrong either with how I set up the virtual authenticator (line 15 above) or create a credential (utils.sub.js)?

Reply:
I was able to reproduce it without SPC, so I think it's a WebAuthn issue (https://github.com/web-platform-tests/wpt/pull/31602). I'll continue to bisect options on that CL to see if I can figure it out. Unfortunately getting a personal build of Chromium hooked up to WPT is a little non-trivial (not impossible, just tedious).

Reply:
Oh, no, I was wrong - it's probably SPC. I had forgotten that I was still specifying the `payment` extension. Without that, the create-credential issue doesn't reproduce.

I expect that some SPC code is failing to check the virtual authenticator override properly, and so when there isn't a valid authenticator on the platform (such as on CI) it fails. This is good, because I should be able to reproduce and dig in on my linux box then :)

Reply:
Figured it out - compounding errors :D. SPC isn't enabled on Linux, but we have a bug where blink thinks it is and the browser thinks it isn't, so the browser just drops the mojo message... fun. Sorry for the noise here!